scripts/analyze

```diff
@@ -0,0 +1,298 @@
+            feature = jrs.read()
+        with open(args['PROFILE'], "w") as p:
+            for i in list:
+                feature = feature.replace(i, "#" + i)
+            p.write(feature)
+
+def write_makefile(src, zconf, file):
+    with open(args['MAKEBASE'], "r") as b:
+        base = b.read()
+    with open(file, "w") as f:
+        f.write(base)
+        if src:
+            f.write("obj-y += \\\n")
+            for i in src:
+                f.write("\t" + i.replace(".c", ".o") + " \\\n")
+            f.write("\nccflags-y += \\\n")
+            for i in zconf:
+                f.write("\t" + i + " \\\n")
+            f.write("\n")
+
+def write_zconf(list, file):
+    with open(file, "w") as f:
+        for i in list:
+            f.write(i + '\n')
+
+def write_modules(list, tree):
+    headers = []
+    inits = {}
+    gbl_inits = {}
+    with open('outdir/include/zjs_modules_gen.h', 'w') as file:
+        for i in list:
+            entry = tree.get(i, None)
+            if entry:
+                if 'init' in entry:
+                    inits[i] = {}
+                    inits[i]["inits"] = entry['init']
+                if 'cleanup' in entry:
+                    inits[i]["cleanups"] = entry['cleanup']
+                if 'header' in entry:
+                    merge_lists(headers, entry['header'])
+                if 'global_init' in entry:
+                    gbl_inits[i] = {}
+                    gbl_inits[i]["inits"] = entry['global_init']
+                if 'global_cleanup' in entry:
+                    gbl_inits[i]["cleanups"] = entry['global_cleanup']
+        # Add include headers
+        file.write("// This is a generated file\n")
+        file.write("// Includes:\n")
+        file.write("#include \"jerryscript.h\"\n")
+        for i in headers:
+            file.write("#include \"%s\"\n" % i)
+        file.write("""typedef jerry_value_t (*initcb_t)();
+typedef void (*cleanupcb_t)();
+typedef struct module {
+    const char *name;
+    initcb_t init;
+    cleanupcb_t cleanup;
+    jerry_value_t instance;
+} module_t;
+typedef struct gbl_module {
+    const char *name;
+    void (*init)();
+    void (*cleanup)();
+} gbl_module_t;
+module_t zjs_modules_array[] = {\n""")
+        for i in inits:
+            file.write("    {\"%s\", %s" % (i, inits[i]["inits"][0]))
+            if 'cleanups' in inits[i]:
+                file.write(", %s" % inits[i]["cleanups"][0])
+            file.write("},\n")
+        file.write("};\n")
+        file.write("gbl_module_t zjs_global_array[] = {\n")
+        for i in gbl_inits:
+            file.write("    {\"%s\", %s" % (i, gbl_inits[i]["inits"][0]))
+            if 'cleanups' in gbl_inits[i]:
+                file.write(", %s" % gbl_inits[i]["cleanups"][0])
+            file.write("},\n")
+        file.write("};\n")
+
+if args['BOARD'] == 'linux':
+    restrict = args['RESTRICT'].split(',')
+    restrict = [i.strip() for i in restrict]
```

Comment:
Make was adding a space between them if I escaped newline like:

```
RESTRICT=foo,\
bar,\
baz
```

Your right it shouldn't do that but this is why I did it.